Apps/phbitglass/bitglass_connector.py

```diff
@@ -0,0 +1,779 @@
+
+        # everything else is actually an error at this point
+        message = "Can't process response from server. Status Code: {0} Data from server: {1}".format(
+            r.status_code,
+            r.text.replace('{', '{{').replace('}', '}}')
+        )
+
+        return RetVal(action_result.set_status(phantom.APP_ERROR, message), None)
+
+    def _make_rest_call(self, url, endpoint, action_result, method="get", **kwargs):
+        # **kwargs can be any additional parameters that requests.request accepts
+
+        resp_json = None
+
+        try:
+            resp_json, r = bgapi().RestCall(endpoint, kwargs['params'])
+        except requests.exceptions.ConnectionError:
+            err_msg = 'Error Details: Connection Refused from the Server'
+            return RetVal(action_result.set_status(phantom.APP_ERROR, err_msg), resp_json)
+        except Exception as ex:
+            err_msg = self._get_error_message_from_exception(ex)
+            return RetVal(
+                action_result.set_status(
+                    phantom.APP_ERROR, "Error Connecting to server. Details: {0}".format(err_msg)
+                ), resp_json
+            )
+
+        return self._process_response(r, action_result)
+
+    def _callBitglassApi(self, action_result, _type, action, param, params):
+        if params:
+            # Make rest call
+            url, endpoint = bgapi().RestParamsConfig(None, '1', _type, action)
+            ret_val, response = self._make_rest_call(url, endpoint, action_result, params=params, headers=None)
+            if phantom.is_fail(ret_val):
+                return action_result.get_status()
+
+            # Add JSON data
+            action_result.add_data(response)
+
+        # Return success, no need to set the message, only the status
+        # BaseConnector will create a textual message based off of the summary dictionary
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    def _handle_test_connectivity(self, param):
+        # Add an action result object to self (BaseConnector) to represent the action for this param
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        # NOTE: test connectivity does _NOT_ take any parameters
+        # i.e. the param dictionary passed to this handler will be empty.
+        # Also typically it does not add any data into an action_result either.
+        # The status and progress messages are more important.
+
+        # This code is similar to on_poll() with overriding logtypes with cloudsummary and all the data descarded
+
+        # HACK Skip the actual API call to poll by setting APP_SUCCESS if needed
+        ret_val, _ = RetVal(action_result.set_status(phantom.APP_ERROR), {})
+        if phantom.is_fail(ret_val):
+            # Multiple rest requests below, each one containing multiple log events
+            try:
+                status = bgapi(self).PollLogs(conf, [u'cloudsummary'])
```

Comment:
The app is in Python version 3,  but why it's using Unicode type? If there are no specific requirements, we can use string as per python3's standard practice in all the files.

@asingh-bitglass, could you please help us with this?

Reply:
```suggestion
                status = bgapi(self).PollLogs(conf, ['cloudsummary'])
```

Reply:
This is because some of this code is required to run on Python 2. We are planning on removing the u'' notation (as well as the other Python 2 support things like import six etc.) as soon as Python 2 support is no longer necessary (a matter of a few months). Can you please grant us a waiver on those in the meantime, thanks

Reply:
Okay, no worries. For now, we can move forward but it would be good if you can ensure that you didn't encounter any issues running as a py3 app.

Reply:
Yep, py3 is absolutely supported. It was ported many months ago and tested ever since. We'll address this on the next submission, thanks